src/webglimpse/timeline/timeline_events_row.ts

```diff
@@ -705,39 +710,87 @@ module Webglimpse {
                 '                                                                ',
                 '  void main( ) {                                                ',
                 '      gl_Position = vec4( ( -1.0 + 2.0*a_XyFrac ), 0.0, 1.0 );  ',
                 '      v_Color = a_Color;                                        ',
+                '      v_relativeXy = a_relativeXy;                                ',
+                '      v_fillPattern = a_fillPattern;                                ',                
                 '  }                                                             ',
                 '                                                                '
             );
 
-        var program = new Program( xyFrac_vColor_VERTSHADER, varyingColor_FRAGSHADER );
+
+         var fillPattern_FRAGSHADER = Webglimpse.concatLines(
+                '                                                                           ',
+				'  precision highp float;                                                   ',
+				'  varying vec4 v_Color;                                                    ',
+   				'  varying vec2 v_relativeXy;                                                ',
+   				'  varying float v_fillPattern;                                              ',                   
+				'  void main( ) {                                                           ',
+                '       if(v_fillPattern==0.){                                               ', // If fillPattern is solid, set the color and return
+				'           gl_FragColor = v_Color;                                         ',
+                '           return;                                                         ',
+                '       }                                                                   ',
+				'       float pi = 3.14159265359;                                           ',
+				'       float wave = sin(32.*2.*pi*(v_relativeXy.x+v_relativeXy.y*0.05));           ',
+				'       wave = (wave+1.)/2.;                                                ',
+				'       if(wave>0.95){                                                      ',
+				'           gl_FragColor = v_Color;                                         ',	
+				'       }else if(wave>0.4) {                                                ',
+				'           gl_FragColor = (wave + 0.3)*v_Color;                            ', // Add 0.3 to wave to reduce the dark outline of the stripes
+				'       } else {                                                            ',
+                '           gl_FragColor = vec4(1.);                                        ', // Make the background white
+                '       }	                                                                ',
+				'  }                                                                        ',
+				'                                                                           ' 
```

Comment:
I think it's much more natural to have the v_relativeXy work in pixels. So the initialization of the v_relativeXy values becomes:

```
var yHeightPixels = jTop - jBottom;
indexRelativeXys = putQuadXys(relativeXys, indexRelativeXys, 0.0, xWidthPixels, 0.0, yHeightPixels );
```

Then, the shader could be something like (for vertical stripes):

```
if ( mode ( v_relativeXy.x, 10.0 ) < 5.0 ) {
    gl_FragColor = v_Color;
}
else {
    gl_FragColor = vec4( 1., 1., 1., 1. );
}
```


Or (for diagonal stripes):

```
if ( mode ( v_relativeXy.x * 0.2 * v_relativeXy.y, 10.0 ) < 5.0 ) {
    gl_FragColor = v_Color;
}
else {
    gl_FragColor = vec4( 1., 1., 1., 1. );
}
```

The parameters (10 and 5 indicating 5 pixels of one color then 5 of the other, and 0.2 indicating the slope of the line) have nice understandable physical meaning that way as well. Those parameters could even be defined as uniform variables in the shader and set on a per-event basis by TimelineEventModel fields to allow easy customization of the event pattern.


Finally, we need to add anti-aliasing in the form of color blending between the two stripe colors to get rid of the jagged edges on the stripes:

```
// all these constants probably should be shader uniform variables
// for maximum customizability

// the width in pixels of the first color stripe
float stripeWidth1 = 5.0;
// the width in pixels of the second color stripe
float stripeWidth2 = 5.0;
// the slant of the stipes: 0 = horizontal, 1 = 45 degrees
float slant = 0.2;
float stripeWidthTotal = stripeWidth1 + stripeWidth2;
// the width in pixels of the feathering/antialiasing between stripes
float featherWidth = 1.0;
// the color of the second stripe
vec4 v_Color2 = vec4( 1., 0., 0., 1. );

// calculate the value indicating where we are in the stripe pattern
float stripeCoord = mod( v_relativeXy.x + slant * v_relativeXy.y , stripeWidthTotal );

// we are in the feather region beween the two stripes
if ( stripeCoord < featherWidth ) {
    float diff = stripeCoord / featherWidth;
    gl_FragColor = vec4 ( v_Color.xyz * diff + (1.0-diff) * v_Color2.xyz, 1.0 );
}
// we are in the color 1 stripe
else if ( stripeCoord < stripeWidth1 ) {
    gl_FragColor = v_Color;
}
// we are the feather region berween the two stripes
else if ( stripeCoord  < stripeWidth1 + featherWidth ) {
    float diff = ( stripeCoord - stripeWidth1 ) / featherWidth;
    gl_FragColor = vec4 ( v_Color.xyz * (1.0-diff) + diff * v_Color2.xyz, 1.0 );
}
// we are in the color 2 stripe
else {
    gl_FragColor = v_Color2;
}
```